lib/Server.js

```diff
@@ -1963,7 +1963,49 @@ class Server {
       const _path = req.path;
 
       try {
-        const filename = this.middleware.getFilenameFromUrl(`${_path}.js`);
+        let filename;
+        let extension;
+        let scriptType;
+
+        // _path = "/main" -> outputFilename = "main"
+        // _path = "/main.other" -> outputFilename = "main.other"
+        const outputFilename = _path.slice(1);
+
+        let isOutputModule = false;
+
+        if (typeof this.compiler.compilers !== "undefined") {
+          this.stats.stats.forEach((stats) => {
+            stats.compilation.assetsInfo.forEach((value, key) => {
+              const assetExtension = path.extname(key);
+              const assetBasename = path.basename(key, assetExtension);
+
+              if (assetBasename === outputFilename) {
+                extension = assetExtension;
+                isOutputModule = value.javascriptModule;
+              }
+            });
+          });
```

Comment:
Using `forEach` is slow, better to move this in `const findInfo = () => { for (const item of stats) { for (const assetsInfo of item.compilation.assetsInfo) { if (one === two) {  return { extension, isModule } } } }}` (pseudo code)